src/Command/Checker.php

```diff
@@ -47,18 +49,32 @@ protected function configure()
                 'directory',
                 InputArgument::IS_ARRAY | InputArgument::REQUIRED,
                 'Directory to scan *.php files'
+            )
+            ->addOption(
+                'exclude-dir',
+                null,
+                InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+                'Prevent directory to be used when check for doc headers'
+            )
+            ->addOption(
+                'exclude',
+                null,
+                InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+                'Prevent file to be used when check for doc headers'
             );
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $directory = $input->getArgument('directory');
-        $finder    = (new IOResourcePathResolution($directory))->__invoke();
-        $validator = new RegExp($this->header);
+        $directory           = $input->getArgument('directory');
+        $excludedDirectories = $input->getOption('exclude-dir') ?: [];
+        $excludedFiles       = $input->getOption('exclude') ?: [];
+        $finder              = (new IOResourcePathResolution($directory, $excludedDirectories, $excludedFiles))->__invoke();
+        $validator           = new RegExp($this->header);
 
         /* @var $file \Symfony\Component\Finder\SplFileInfo */
-        foreach ($finder as $directory) {
-            foreach ($directory as $file) {
+        foreach ($finder as $directoryList) {
+            foreach ($directoryList as $file) {
```

Comment:
@weierophinney it had a previous variable called directory, I was wrongly overriding its content it here. what do you suggest?